app/models.py

```diff
@@ -88,6 +88,22 @@ def is_authenticated(user_by_email: User, password: User) -> bool:
 
         return check_password_hash(user_by_email.password, password)
 
+    @staticmethod
+    def get_feed_by_user_id(user_id: int) -> List[Post]:
+        query = f'''
+            SELECT post_id
+            FROM post
+            JOIN followers f ON post.user_id = f.following_id
+            WHERE {user_id} = f.user_id
+            ORDER BY post.created_at DESC
+            LIMIT 50
```

Comment:
I think it would be best we don't limit the feed to max 50 recent posts. Implementing lazy loading will be very simple on the front end so better off doing it now then worrying about it later.

Reply:
for FE lazy loading we will need to make an endpoint with a post request which we can hit from the front end as the user scrolls down the page. we would still need a limit of some number, but we would paginate the database request.

Reply:
or maybe i am misunderstanding what you're saying?

Reply:
I mean pagination. We could keep a hidden offset number in the UI and pass that to the back end to be used for the next 50 oldest posts from followers. From there return the rendered html to the front end and insert it at the end of the page.

Reply:
Ok there is a new endpoint now that will get the next offset of posts. The offset is handled in the session so we dont have to worry about that in the FE.
Standard flow would be make the regular get to the home page -> as user scrolls make the get to '/next-feed' which will return back the html then you can just append it directly in js.
All you will have to do to get it working on the backend side is once you have your feed posts template made, update the service function `get_feed_with_offset` render_Template call so instead of passing the list of posts into the feed.html it passes it into a template that you can directly append.

Reply:
Good with me.